### PR TITLE
fix subsample bug

### PR DESCRIPTION
**Pull request by cdebourcy**
Set subsampled_reads correctly if subsample > remaining_reads

### CODE REVIEW

File: app/models/pipeline_run.rb [hunk 1]
```diff
@@ -269,7 +269,9 @@ def update_genera
 
   def subsampled_reads
     return remaining_reads unless subsample
-    subsample * sample.input_files.count
+    result = subsample * sample.input_files.count
```
[the remark this answers is not in the document]
Comment (cdebourcy) [reply app/models/pipeline_run.rb]: subsample is the number of read pairs to subsample to (1_000_000 default).
subsample_fraction is the fraction of non-host reads that actually went into the report, i.e. if total number of read pairs was 2_000_000, it is 0.5

File: app/models/pipeline_run.rb [hunk 2]
```diff
@@ -269,7 +269,9 @@ def update_genera
...
+    return remaining_reads unless result < remaining_reads
+    result
```
Comment (yunfangjuan) [inline app/models/pipeline_run.rb]: (remaining_reads < result)? remaining_reads: result

File: app/models/pipeline_run.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (yunfangjuan) [reply app/models/pipeline_run.rb]: OK. prolly should document this function better.

**Comment by yunfangjuan** [comment]
OK. I tested this. It works.

**Pull request merged**